srml/support/src/dispatch.rs

```diff
@@ -922,16 +924,16 @@ macro_rules! decl_module {
 					$fn_vis fn $fn_name (
 						$from $(, $param_name : $param )*
 					) $( -> $result )* {
-						// Check doughnut grants permission before dispatching the call
-						let doughnut_opt: Option<$crate::dispatch::Doughnut<$trait_instance::AccountId, $trait_instance::Signature>> = $crate::storage::unhashed::get(b":doughnut");
-
-						if let Some(doughnut) = doughnut_opt {
-							if doughnut.certificate.permissions.is_empty() {
-								return Err("Doughnut with empty permissions");
-							}
-							if doughnut.certificate.permissions[0].0.as_slice() != env!("CARGO_PKG_NAME").as_bytes() {
-								return Err("Doughnut does not grant permission for this domain");
-							}
+						use $crate::dispatch::DoughnutVerifier;
+						// Check if a doughnut exists in this execution context and whether it grants permission to
+						// dispatch the call.
+						if let Some(doughnut) = $crate::storage::unhashed::get(b"doughnut") {
```

Comment:
Let's move the doughnut existential condition and surfacing to a proxy class.
Reasoning: These steps are needed prior to `verify_doughnut` calls, so let's save ourselves some work and write it once.

Reply:
After some thinking I don't want to tackle the proxy class until we start with contract integration otherwise it may be throwaway work